### PR TITLE
Updates for vhd appliance build

### PR DESCRIPTION
**Pull request by JPrause**
Changes to support creating vhd appliance

### CODE REVIEW

**Comment by JPrause** [comment]
@jrafanie @carbonin @abellotti @simaishi 
Please review.

Also, for RHEL7, we needed to add the following the dracut:
hv_storvsc hid_hyperv hv_netvvc hv_vmbus
Not sure if this needed on centos,..but if so, can we add that to base.ks.erb for all builds, or separate out to a partial in includes.ks.erb?

**Comment by Fryguy** [comment]
Syntax error...looks like you missed a comma in a Hash.

File: scripts/spec/target_spec.rb [hunk 1]
```diff
@@ -13,11 +13,11 @@
   end
 
   it "#imagefactory_type" do
-    expect(described_class.new("openstack").imagefactory_type).to eql "openstack-kvm"
+    expect(described_class.new("vhd").imagefactory_type).to eql "vpc"
   end
 
   it "#file_extension" do
-    expect(described_class.new("openstack").file_extension).to eql "qc2"
+    expect(described_class.new("vhd").file_extension).to eql "vpc"
```
Comment (Fryguy) [inline scripts/spec/target_spec.rb]: Why were these tests changed?  Not that it *really* matters, but I was curious why you changed it.

Comment (carbonin) [inline scripts/spec/target_spec.rb]: Why did we change this?  Seems like this was testing the `file_extension` method and wasn't specific to a particular target type.  Same with `imagefactory_type`.

Comment (jrafanie) [inline scripts/spec/target_spec.rb]: I suggested it as a way to verify the changes worked for vhd.

**Comment by Fryguy** [comment]
LGTM.  Can you squash in the syntax error?  Also, have you tested with an actual build?  If not, can you?  @jrafanie or @abellotti can help you out if needed.

**Comment by JPrause** [comment]
Waiting to run a test build.

**Comment by jrafanie** [comment]
> Also, for RHEL7, we needed to add the following the dracut:
hv_storvsc hid_hyperv hv_netvvc hv_vmbus
Not sure if this needed on centos,..but if so, can we add that to base.ks.erb for all builds, or separate out to a partial in includes.ks.erb?

@carbonin @abellotti what about this part?

@JPrause was there anything else you had to change in the kickstart besides the `dracut` line?

**Comment by JPrause** [comment]
@jrafanie Not that I'm aware of,...the dracut additions may not be needed for centos. We're trying ot create a test upstream build now. 1st hurdle create vhd build,...2nd will it boot, 3rd does it look pretty.

**Comment by simaishi** [comment]
@JPrause I thought testing proved that modifying ifcfg-eth0 and removing 70-persistent-net.rules were also required for RHEL7?

**Comment by JPrause** [comment]
and yes,..there's that as well.

**Comment by carbonin** [comment]
Test failed.  It looks like that machine is missing a plugin?
```
2015-10-20 14:48:12,877 ERROR imgfac.Builder.Builder thread(b78d1c44) Message: Exception encountered in _customize_image_for_target thread
2015-10-20 14:48:12,877 ERROR imgfac.Builder.Builder thread(b78d1c44) Message: No plugin .info file loaded for target: vpc
Traceback (most recent call last):
  File "/build/imagefactory/imgfac/Builder.py", line 226, in _customize_image_for_target
    self.cloud_plugin = plugin_mgr.plugin_for_target(target)
  File "/build/imagefactory/imgfac/PluginManager.py", line 140, in plugin_for_target
    raise ImageFactoryException("No plugin .info file loaded for target: %s" % (target))
ImageFactoryException: No plugin .info file loaded for target: vpc

```

**Comment by jrafanie** [comment]
> est failed. It looks like that machine is missing a plugin?
> 
> 2015-10-20 14:48:12,877 ERROR imgfac.Builder.Builder thread(b78d1c44) Message: Exception encountered in _customize_image_for_target thread
> 2015-10-20 14:48:12,877 ERROR imgfac.Builder.Builder thread(b78d1c44) Message: No plugin .info file loaded for target: vpc
> Traceback (most recent call last):
>   File "/build/imagefactory/imgfac/Builder.py", line 226, in _customize_image_for_target
>     self.cloud_plugin = plugin_mgr.plugin_for_target(target)
>   File "/build/imagefactory/imgfac/PluginManager.py", line 140, in plugin_for_target
>     raise ImageFactoryException("No plugin .info file loaded for target: %s" % (target))
> ImageFactoryException: No plugin .info file loaded for target: vpc

I wonder if there's some packages we need to install on the imagefactory vm, see: https://github.com/ManageIQ/manageiq-appliance-build#setup-for-vsphere-plugin for each provider...

@JPrause do you know or can ask what version of imagefactory it worked with and if there are any dependencies we need to install?

**Comment by simaishi** [comment]
Got an image built by converting raw image to vhd, moving on...

**Pull request closed without merging**